server/lib/cordova.js

```diff
@@ -0,0 +1,215 @@
+/* global Push, SystemLogger */
+
+Meteor.methods({
+	log() {
+		return console.log(...arguments);
+	},
+
+	push_test() {
+		const user = Meteor.user();
+
+		if (!user) {
+			throw new Meteor.Error('error-not-allowed', 'Not allowed', {
+				method: 'push_test'
+			});
+		}
+
+		if (!RocketChat.authz.hasRole(user._id, 'admin')) {
+			throw new Meteor.Error('error-not-allowed', 'Not allowed', {
+				method: 'push_test'
+			});
+		}
+
+		if (Push.enabled !== true) {
+			throw new Meteor.Error('error-push-disabled', 'Push is disabled', {
+				method: 'push_test'
+			});
+		}
+
+		const query = {
+			$and: [{
+				userId: user._id
+			}, {
+				$or: [{
+					'token.apn': {
+						$exists: true
+					}
+				}, {
+					'token.gcm': {
+						$exists: true
+					}
+				}]
+			}]
+		};
+
+		const tokens = Push.appCollection.find(query).count();
+
+		if (tokens === 0) {
+			throw new Meteor.Error('error-no-tokens-for-this-user', 'There are no tokens for this user', {
+				method: 'push_test'
+			});
+		}
+
+		Push.send({
+			from: 'push',
+			title: `@${user.username}`,
+			text: TAPi18n.__('This_is_a_push_test_messsage'),
+			apn: {
+				text: `@${user.username}:\n${TAPi18n.__('This_is_a_push_test_messsage')}`
+			},
+			sound: 'default',
+			query: {
+				userId: user._id
+			}
+		});
+
+		return {
+			message: 'Your_push_was_sent_to_s_devices',
+			params: [tokens]
+		};
+	}
+});
+
+function sendPush(service, token, options, tries = 0) {
+	const data = {
+		data: {
+			token: token,
+			options: options
+		}
+	};
+
+	return HTTP.post(RocketChat.settings.get('Push_gateway') + `/push/${service}/send`, data, function(error, response) {
+		if (response && response.statusCode === 406) {
+			console.log('removing push token', token);
+			Push.appCollection.remove({
+				$or: [{
+					'token.apn': token
+				}, {
+					'token.gcm': token
+				}]
+			});
+			return;
+		}
+
+		if (!error) {
+			return;
+		}
+
+		SystemLogger.error(`Error sending push to gateway (${tries} try) ->`, error);
+
+		if (tries <= 6) {
+			const milli = Math.pow(10, tries + 2);
+
+			SystemLogger.log('Trying sending push to gateway again in', milli, 'milliseconds');
+
+			return Meteor.setTimeout(function() {
```

Comment:
Here we see that 27h retry again, like that from Integrations... Maybe also use that proposed setting here?